test/testsymboldatabase.cpp

```diff
@@ -2298,6 +2299,19 @@ class TestSymbolDatabase : public TestFixture {
         ASSERT(f && f->function() && !f->function()->isExtern() && f->function()->retDef->str() == "void");
     }
 
+    void externalFunctionsInsideAFunction() {
+        GET_SYMBOL_DB("void foo( void )\n"
+                      "{\n"
+                      "    extern void bar( void );\n"
+                      "    bar();\n"
+                      "}\n");
+
+        ASSERT(db && errout.str().empty());
+
+        const Token *f = Token::findsimplematch(tokenizer.tokens(), "bar");
+        ASSERT(f && f->function() && f->function()->isExtern() && f->function()->retDef->str() == "void");
```

Comment:
The condition `f == f->function()->tokenDef` is also true as far as I know please add that check here..